### PR TITLE
Update to .NET 8

### PR DESCRIPTION
**Pull request by janusw**
This PR updates Maui.Tabs and MauiSample from .NET 6 to .NET 8 (since .NET MAUI 6 is out of support by now). It fixes all errors I encountered in the process (and a few obsolescence warnings).

It enables me to build the sample again, thereby fixing #115.

IMHO it is also a first step that should be taken before targeting .NET 9, as proposed in #118.

### CODE REVIEW

**Comment by janusw** [comment]
Too bad there is no CI here to verify the build ...

(I could contribute some hooks for GitHub Actions, if this is desired.)

**Comment by roubachof** [comment]
Thank you for your proposal, I sent you an invite

File: MauiSample/Presentation/Views/TabA.xaml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
             <Style x:Key="MaterialTabStyle" TargetType="tabs:MaterialUnderlinedTabItem">
                 <Setter Property="SelectedTabColor" Value="{StaticResource Primary}" />
-                <Setter Property="SelectedTabTextColor" Value="{StaticResource Secondary}" />
+                <!--<Setter Property="SelectedTabTextColor" Value="{StaticResource Secondary}" />-->
```
Comment (roubachof) [inline MauiSample/Presentation/Views/TabA.xaml:14]: why is this commented ?

Comment (janusw) [reply MauiSample/Presentation/Views/TabA.xaml:14]: To fix a build error:
```
Cannot resolve property "SelectedTabTextColor"
on type "MaterialUnderlinedTabItem
(property missing or missing accessors)".
```

Comment (janusw) [reply MauiSample/Presentation/Views/TabA.xaml:14]: AFAICS the error message is correct, right? `SelectedTabTextColor` is only available in `BottomTabItem`, but not in `MaterialUnderlinedTabItem` (both inherit from `TabTextItem`, directly or indirectly).
Not sure why this was missed with net6.0. Probably improved diagnostics in net8.0, I guess ...

**Comment by janusw** [comment]
> Too bad there is no CI here to verify the build ...
> 
> (I could contribute some hooks for GitHub Actions, if this is desired.)

See 
260dd7192921d115f82de822f4786b2d7c70ac30

**Comment by roubachof** [comment]
Thank you so much for this PR !

**Comment by roubachof** [comment]
I guess it fixes https://github.com/roubachof/Sharpnado.Tabs/issues/116 right ?

**Comment by janusw** [comment]
> Thank you so much for this PR !

Thanks for merging!


> I guess it fixes #116 right ?

I'm not sure (have not tried it with .NET 9 yet). It sounds like #116 is caused by references to obsoleted code (`Device` and `Frame`). I fixed the obsolescence warnings caused by `Device`, but not those related to `Frame` (since it is still valid in MAUI 8 and only obsoleted in MAUI 9).

Since `Frame` is only obsoleted (but not yet removed) in .NET 9, I don' t think it should cause any runtime errors. So the issue might be fixed after all.

@daltzctr Could you please check if this PR fixes #116 for you?

**Pull request merged**